Use consistent tone in overview

mpkorstanje opened the pull request:
- Both the Gherkin and the Cucumber section follow the Q&A pattern.
  For consistency the Step Definition section should follow this
  pattern.

- Step Definition section referenced feature files. A concept not
  introduced prior. Change this to reference Gherkin steps in instead
  (which have been introduced before)

- Diagram was missing. Added some ASCII art.

- Removed BDD snark from introduction.

- Inlined the prompt to read about BDD first. This is the first thing users will read.
No need to preempt it with a notice before they've even started.

- There seems to be a bit of a contradiction between these two pages:

  https://cucumber.io/docs/guides/overview/#what-is-gherkin
   - developers write Gherkin 
   - testers or developers if used as a test automation tool.
   - not by product owners and business analysts write Gherk

  https://cucumber.io/docs/bdd/who-does-what/
   - initially written by a domain expert.
   - reviewed & edited by devs
   - reviewed by domain expert

  Removing the notes from the intro resolves the contradiction in favour of
the more nuanced bdd introduction.

mpkorstanje committed:
Use consistent tone in overview

- Both the Gherkin and the Cucumber section follow the Q&A pattern.
  For consistency the Step Definition section should follow this
  pattern.

- Step Definition section referenced feature files. A concept not
  introduced prior. Change this to reference Gherkin steps in instead
  (which have been introduced before)

- Diagram was missing. Added some ASCII art.

olleolleolle commented on content/docs/guides/overview.md:
````diff
@@ -50,30 +50,34 @@ Gherkin serves multiple purposes:
-<!-- TODO: Illustration (Feature) - (Step Defs) -> (System) -->
+```
+ | Steps in Gherkin | --matched with--> | Step Definitions | --manipulates--> | System |
+```
````
```
┌────────────┐                 ┌──────────────┐                 ┌───────────┐
│   Steps    │                 │     Step     │                 │           │
│ in Gherkin ├──matched with──▶│ Definitions  ├───manipulates──▶│  System   │
│            │                 │              │                 │           │
└────────────┘                 └──────────────┘                 └───────────┘
```

I opened Monodraw, and had so much fun drawing this.

mpkorstanje replied:
Thanks! Much appreciated!

mpkorstanje committed:
Link introduction on main page

mpkorstanje committed:
Inline the prompt to read BDD. This is the first thing users will read.
No need to preempt it with a notice.

mpkorstanje committed:
Better ascii art

mpkorstanje committed:
There seems to be a bit of a contradiction between these two pages:

https://cucumber.io/docs/guides/overview/#what-is-gherkin
 - developers write Gherkin
 - testers or developers if used as a test automation tool.
 - not by product owners and business analysts write Gherk

https://cucumber.io/docs/bdd/who-does-what/
 - initially written by a domain expert.
 - reviewed & edited by devs
 - reviewed by domain expert

Removing the note from the intro resolves the contradiction in favour of
the more nuanced bdd introduction.

mpkorstanje committed:
Move intro to the intro above the header

mlvandijk commented on content/docs/guides/overview.md:
````diff
@@ -50,30 +45,25 @@ Gherkin serves multiple purposes:
+│ in Gherkin ├──matched with──▶│ Definitions  ├───manipulates──▶│  System   │
+│            │                 │              │                 │           │
+└────────────┘                 └──────────────┘                 └───────────┘
+```
````
Thanks for adding an illustration; this should be super helpful. (An actual picture might be better, but a diagram is great in the mean time)

mlvandijk commented on content/docs/guides/overview.md:
````diff
@@ -3,22 +3,17 @@ title: Introduction
+that the software does what those specifications say. The specifications
+consists of multiple *examples*, or *scenarios*. For example:
 
 ```gherkin
````
Tbh I have no idea what this example means. Could we replace it with something people would (more) easily recognise?

mlvandijk reviewed (commented):
Great edit, thank you! Some minor suggestions, but don't let those stop you from merging. We can always take them up in another iteration.
